script/make_release.py

```diff
@@ -0,0 +1,392 @@
+
+                LOGGER.info('{}: Verifying bench.'.format(arch))
+
+                bench = run_bench(src_file)
+                if bench == expected_bench:
+                    LOGGER.info('{}: Correct bench: {}'.format(arch, bench))
+                    LOGGER.info('{}: Producing the release archive.'.format(arch))
+                    dest_filename = 'stockfish{}-{}{}'.format(prefix, arch, EXE_SUFFIX)
+                    pack_binary_with_source(out_dir, repo_dir, src_file, dest_filename)
+                else:
+                    LOGGER.error('{}: Invalid bench. Expected {}, got {}.'.format(arch, expected_bench, bench))
+
+
+            LOGGER.info('{}: Removing temporary directory {}'.format(arch, copy_dir.path))
+
+def get_possible_targets(repo_dir):
+    '''
+    Parse the output of `make help` and extract the available targets
+    '''
+    p = subprocess.run(['make', 'help'], cwd=os.path.join(repo_dir, 'src'),
+                       encoding='utf-8', stdout=subprocess.PIPE, universal_newlines=True)
+
+    targets = []
+    read_targets = False
+
+    for line in p.stdout.splitlines():
+        if 'Supported compilers:' in line:
+            read_targets = False
+        if read_targets and len(line.split()) > 1:
+            targets.append(line.split()[0])
+        if 'Supported archs:' in line:
+            read_targets = True
+
+    targets = [target for target in targets if "apple" not in target if "arm" not in target if "ppc" not in target]
+
+    return targets
+
+def main():
+    shutil.rmtree(CLONE_DIR, True, onerror=remove_readonly)
+
+    with TemporaryDirectory(CLONE_DIR, False):
+        try:
+            os.mkdir(OUT_DIR)
+        except FileExistsError as e:
+            LOGGER.warning('Directory {} already exists.'.format(OUT_DIR))
+
+        clone_stockfish(CLONE_DIR, REPO)
+
+        expected_bench = EXPECTED_BENCH if EXPECTED_BENCH is not None else get_bench_from_git_history(CLONE_DIR)
+        if expected_bench is None:
+            LOGGER.error('Could not retrieve bench from git history. Exiting.')
+            return
+        else:
+            LOGGER.info('Expected bench is {}.'.format(expected_bench))
+
+        archs = ARCHS if ARCHS else get_possible_targets(CLONE_DIR)
+
+        if not archs:
+            LOGGER.info('No archs specified. Exiting.')
+            return
+
+        # Compute how many threads each worker can use
+        # Try to find as equal distribution as possible
+        num_workers = min([NUM_THREADS, len(archs), MAX_NUM_WORKERS])
+        num_threads_per_worker = [1] * num_workers
+        for i in range(NUM_THREADS - num_workers):
+            num_threads_per_worker[i % num_workers] += 1
+        num_workers_by_thread_id = dict()
+        lock = threading.Lock()
+
+        def run_task(arch):
+            try:
+                with lock:
+                    thread_id = threading.current_thread().native_id
```

Comment:
`threading.current_thread().native_id` python 3.8

It's running with `threading.current_thread().ident` but please double check, I made a monkey style bugfix :)

https://docs.python.org/3.6/library/threading.html#threading.Thread.ident